File: NOVEMBER-2025/17-11-2025/1437.c
bool kLengthApart(int* nums, int numsSize, int k) {
    int prev = -1;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 1) {
            if (prev != -1 && (i - prev - 1) < k) {
                return false;
            }
            prev = i;
        }
    }
    return true;
}

File: NOVEMBER-2025/17-11-2025/1437.kt
class Solution {
    fun kLengthApart(nums: IntArray, k: Int): Boolean {
        var prev = -1
        for (i in nums.indices) {
            if (nums[i] == 1) {
                if (prev != -1 && (i - prev - 1) < k) {
                    return false
                }
                prev = i
            }
        }
        return true
    }
}